solutions/silver/cf-104002E.mdx

```diff
@@ -0,0 +1,74 @@
+---
+id: cf-104002E
+source: CF
+title: William and Robot
+author: Alex Du
+---
+
+## Explanation
+
+First note that for any even number $k \leq n$, a valid solution must have selected at most $k/2$ integers. This is the only requirement for a valid solution.
+
+To solve this problem, we can use a greedy algorithm. For any even $k$, if we know the $k/2$ integers that make the highest sum, we can use this to find the $k/2+1$ integers for $k+2$.
+
+The base case is $k = 2$, where the highest sum is simply the max of the first two integers.
+
+Similarly, we can always select the max of the $k-1$ and $k-2$ elements, since we will still be one step ahead of the robot.
```

Comment:
```suggestion
Similarly, we can always select the maximum of the $k-1$ and $k-2$ elements, since we will still be one step ahead of the robot.
```

Reply:
`since we will still be one step ahead of the robot.`

doesn't really make sense